docs/validation.md

```diff
@@ -616,21 +616,220 @@ While this behaviour is unexpected (issues [#129](https://github.com/ajv-validat
+console.log(data1); // { foo: "A", a: "" }
+
+
+const data2 = {
+  foo: "",
+  a: "", 
+  b: 1 // will NOT be removed; `unevaluatedProperties` != false
+}
+
+console.log(validate(data2)) // true
+console.log(data2); // { foo: "A", a: "", b: 1 }
+
+```
+
+Example 2 : remove unevaluated properties with combination keyword
+
+```javascript
+
+const ajv = new Ajv({removeUnevaluated: true})
+
+const schema = {
+  $schema: "http://json-schema.org/draft/2019-09/schema#",
+  type: "object",
+  properties: {
+    foo: {
+      type: "object",
+      oneOf: [
+        {
+          properties: {
+            x: { type: "string" }
+          },
+          required: ["x"],
+          unevaluatedProperties: false
+        },
+        {
+          properties: {
+            y: { type: "integer" }
+          },
+          required: ["y"],
+          unevaluatedProperties: true
+        }
+      ]
+    }
+  },
+  required: ["foo"]
+};
+
+const validate = ajv.compile(schema)
+
+const data1 = {
+  foo: {
+    x: "", 
+    y: 1 // will be removed; `unevaluatedProperties` == false
+  }
+}
+
+console.log(validate(data1)) // true
+console.log(data1); // { foo: {x: "" }}
+
+
+const data2 = {
+  foo: {
+    y: 1,
+    z: "" : // will NOT be removed; `unevaluatedProperties` != false
+  }
+}
+
+console.log(validate(data2)) // true
+console.log(data2); // { foo: { y: 1, z: "" }}
+
+```
+
+Example 3 : remove unevaluated items 
 
-This option modifies original data.
 
-Example:
 
 ```javascript
-//TODO-unevaluated: add example
+
+const ajv = new Ajv({removeUnevaluated: true})
+
+const schema = {
+  $schema: "http://json-schema.org/draft/2019-09/schema#",
+  type: "array",
+  items: [{ const: "foo" }],
+  anyOf: [
+    {
+      items: [true, { const: "bar" }]
+    },
+    {
+      items: [true, true, { const: "baz" }]
+    }
+  ],
+  unevaluatedItems: false
+};
+
+const validate = ajv.compile(schema)
+
+const data = ["foo", "bar", "qux"];
+
+console.log(validate(data)) // true
+console.log(data); // ["foo", "bar"];
+
 ```
 
-//TODO-unevaluated: add description text for the option. See removeAdditional for inspiration
+Example 4 : remove unevaluated items and properties
+
+```javascript
+
+const ajv = new Ajv({removeUnevaluated: true})
+
+const schema = {
+  $schema: "http://json-schema.org/draft/2019-09/schema#",
+  type: "array",
+  anyOf: [
+    {
+      items: [
+        {
+          type: "object",
+          properties: { a: { type: "string" } },
+          unevaluatedProperties: false,
+          required: ["a"]
+        }
+      ]
+    },
+    {
+      items: [
+        {
+          type: "object",
+          properties: { b: { type: "string" } },
+          unevaluatedProperties: true,
+          required: ["b"]
+        }
+      ]
+    }
+  ],
+  unevaluatedItems: false
+};
+
+const validate = ajv.compile(schema)
+
+const data1 = [{ a: "", c: "" }, {b : "", c: ""}, {c: ""}];
+
+console.log(validate(data1)) // true
+console.log(data1); // [{ a: ""}, {b : "", c: ""}];
```

Comment:
this is really the intent of the schema?

An array with anyOf the following:
items where the first object has {a: string}
items where the first object has {b: string, ...props}

Wouldn't we just expect `[{ a: ""}]` in this scenario? or even  `[{ a: ""},  {b : "", c: ""}, {c: ""}]`

Why is this the expected result (I just don't understand)

Reply:
Yes you are right. The first item has unevaluated property c, so c gets removed. The other items are unevaluated items so they are removed from the array.